flow/connectors/postgres/postgres.go

```diff
@@ -106,14 +100,25 @@ func NewPostgresConnector(ctx context.Context, env map[string]string, pgConfig *
 		replConn:               nil,
 		replState:              nil,
 		replLock:               sync.Mutex{},
-		customTypesMapping:     customTypeMap,
+		customTypesMapping:     nil,
 		metadataSchema:         metadataSchema,
 		hushWarnOID:            make(map[uint32]struct{}),
 		logger:                 logger,
 		relationMessageMapping: make(model.RelationMessageMapping),
 	}, nil
 }
 
+func (c *PostgresConnector) lazyInitCustomTypesMapping(ctx context.Context) error {
```

Comment:
this is really error prone unless this method returns the custom types map & code doesn't directly access

Reply:
I'll switch to return map but I'm not sure about the threadsafe requirement, where is that possible? `SetupReplConn` is where I initialize it and that should be before any of the `ReplPing` stuff

Reply:
Mostly pointing out that any call to this function needs to have exclusion on connection, but I don't think that's an issue

Reply:
can tie it to a mutex if needed in the future, same concern for caching version

Reply:
`customTypesMapping` is in some hot paths, keeping it as a field

Reply:
it's okay, nil check is cheap